src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,111 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private int size;
+
+    public MyLinkedList() {
+        head = new Node<>(null, null, null);
+        head.next = head.prev = head;
+        this.size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(value, head, head.prev);
+        newNode.prev.next = newNode;
+        newNode.next.prev = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        Node<T> indNode = (index == size ? head : getNode(index));
+        Node<T> newNode = new Node<>(value, indNode, indNode.prev);
+        newNode.next.prev = newNode;
+        newNode.prev.next = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T a : list) {
+            add(a);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T old = getNode(index).item;
+        getNode(index).item = value;
+        return old;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> element = getNode(index);
+        final T past = element.item;
+        element.prev.next = element.next;
+        element.next.prev = element.prev;
+        size--;
+        return past;
     }
 
     @Override
     public boolean remove(T t) {
+        for (int i = 0; i < size; i++) {
+            if (getNode(i).item == null && t == null
+                    || getNode(i).item.equals(t) && getNode(i).item != null) {
```

Comment:
calling getNode(i) every time is bad for performance, moreover you're calling it three times per iteration 
consider iterate nodes in this method with node = node.next so you need to do it only once